lib/cloud/gcp/vm.go

```diff
@@ -0,0 +1,543 @@
+		}
+		for _, rawInst := range resp {
+			inst := toInstance(rawInst)
+			inst.ProjectID = projectID
+			instances = append(instances, inst)
+		}
+	}
+}
+
+// InstanceRequest contains parameters for making a request to a specific instance.
+type InstanceRequest struct {
+	// ProjectID is the ID of the VM's project.
+	ProjectID string
+	// Zone is the instance's zone.
+	Zone string
+	// Name is the instance's name.
+	Name string
+}
+
+func (req *InstanceRequest) CheckAndSetDefaults() error {
+	if len(req.ProjectID) == 0 {
+		trace.BadParameter("projectID must be set")
+	}
+	if len(req.Zone) == 0 {
+		trace.BadParameter("zone must be set")
+	}
+	if len(req.Name) == 0 {
+		trace.BadParameter("name must be set")
+	}
+	return nil
+}
+
+// getHostKeys gets the SSH host keys from the VM, if available.
+func (clt *instancesClient) getHostKeys(ctx context.Context, req *InstanceRequest) ([]ssh.PublicKey, error) {
+	queryPath := "hostkeys/"
+	guestAttributes, err := clt.InstanceClient.GetGuestAttributes(ctx, &computepb.GetGuestAttributesInstanceRequest{
+		Instance:  req.Name,
+		Project:   req.ProjectID,
+		Zone:      req.Zone,
+		QueryPath: &queryPath,
+	})
+	if err != nil {
+		return nil, trace.Wrap(convertGoogleError(err))
+	}
+	items := guestAttributes.GetQueryValue().GetItems()
+	keys := make([]ssh.PublicKey, 0, len(items))
+	var errors []error
+	for _, item := range items {
+		key, _, _, _, err := ssh.ParseAuthorizedKey([]byte(fmt.Sprintf("%v %v", item.GetKey(), item.GetValue())))
+		if err == nil {
+			keys = append(keys, key)
+		} else {
+			errors = append(errors, err)
+		}
+	}
+	return keys, trace.NewAggregate(errors...)
+}
+
+// GetInstance gets a GCP VM.
+func (clt *instancesClient) GetInstance(ctx context.Context, req *InstanceRequest) (*Instance, error) {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	resp, err := clt.InstanceClient.Get(ctx, &computepb.GetInstanceRequest{
+		Instance: req.Name,
+		Project:  req.ProjectID,
+		Zone:     req.Zone,
+	})
+	if err != nil {
+		return nil, trace.Wrap(convertGoogleError(err))
+	}
+	inst := toInstance(resp)
+	inst.ProjectID = req.ProjectID
+
+	hostKeys, err := clt.getHostKeys(ctx, req)
+	if err == nil {
+		inst.hostKeys = hostKeys
+	} else if !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+	return inst, nil
+}
+
+// SSHKeyRequest contains parameters to add/removed SSH keys from an instance.
+type SSHKeyRequest struct {
+	// Instance is the instance to add/remove keys form.
+	Instance *Instance
+	// User is the user associated with the key.
+	User string
+	// PublicKey is the key to add. Ignored when removing a key.
+	PublicKey ssh.PublicKey
+	// Expires is the expiration time of the key. Ignored when removing a key.
+	Expires time.Time
+}
+
+func (req *SSHKeyRequest) CheckAndSetDefaults() error {
+	if req.Instance == nil {
+		return trace.BadParameter("instance not set")
+	}
+	if req.User == "" {
+		req.User = "teleport"
```

Comment:
Is this always gonna be the same user?
Can you create a constant and use it here?